domains/primitives/certificate-registry/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2021 Subspace Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Primitives for X509 certificate verification
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "std")]
+pub mod host_functions;
+mod runtime_interface;
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+pub use crate::runtime_interface::signature_verification_runtime_interface;
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime_interface::pass_by;
+use sp_runtime_interface::pass_by::PassBy;
+
+/// Signature verification request.
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct SignatureVerificationRequest {
+    /// Der encoded public key info.
+    pub public_key_info: DerVec,
+    /// Der encoded signature algorithm.
+    pub signature_algorithm: DerVec,
```

Comment:
Not necessarily. The algorithm inside the public key info, per standard, mentions on the type of public key. For example for RSA, its `rsaEncyption`. But when it comes to signature verifcation, verifier would also need to know the kind of hash function used. For TLS, this is usually agreed upon when the connection is initialised with server and client. In our case, we would need the sender to provide those details to the runtime so that appropriate hash function is used in this context.